### PR TITLE
Verify server rootCA chain verification in AKO

### PR DESCRIPTION
**Pull request by sudswasavi**
This commit proposes a change that allows AKO to verify the server
side rootca chain if provided by the values.yaml.

This value is rendered in the secret object and subsequently rendered
in the deployment yaml.

NOTE: The rootca must have a proper SAN IP for it to get validated.

### CODE REVIEW

**Comment by sudswasavi** [comment]
@hemantavi FYI for AMKO.

File: pkg/utils/avi_rest_utils.go [hunk 1]
```diff
@@ -50,11 +54,30 @@ func SharedAVIClients() *AviRestClientPool {
 func NewAviRestClientPool(num uint32, api_ep string, username string,
 	password string) (*AviRestClientPool, error) {
 	var p AviRestClientPool
+	ctrlRootCa := os.Getenv("CTRL_ROOTCA")
```
Comment (parimanur) [inline pkg/utils/avi_rest_utils.go:57]: we will have to do a b64decode.

Comment (sudswasavi) [reply pkg/utils/avi_rest_utils.go:57]: I don't think so, Kubernetes does the de-coding and populates the environment variable accordingly. That is how the password/username is also getting handled.

Comment (parimanur) [reply pkg/utils/avi_rest_utils.go:57]: Ok.

**Review by parimanur — APPROVED** [review]

**Review by chauhanshubham — APPROVED** [review]

**Review by shubhamavi — APPROVED** [review]

**Pull request merged**